src/internal/operators/splitBy.ts

```diff
@@ -0,0 +1,99 @@
+import { Observable } from '../Observable';
+import { Subject } from '../Subject';
+import { OperatorFunction } from '../types';
+import { buckets } from './buckets';
+
+export interface SplitByOptions<T> {
+  /**
+   * The factory used to create the subject that will connect the source observable to
+   * multicast consumers.
+   */
+  connector?: (match: boolean) => Subject<T>;
+}
+
+export function splitBy<T, U extends T>(
+  predicate: (value: T) => value is U
+): OperatorFunction<T, [Observable<U>, Observable<Exclude<T, U>>]>;
+export function splitBy<T>(predicate: (value: T) => boolean): OperatorFunction<T, [Observable<T>, Observable<T>]>;
+
+/**
+ * Maps a source observable into an observable that does a single immediate and
+ * synchronous emit that contains an array of two multicasted ("hot")
+ * observables into which the source values are partitioned into according to a
+ * given predicate applied over each source value. The values that match the
+ * predicate are piped into the first observable and the remaining, that do not
+ * match the predicate, into the second one.
+ *
+ * This enables similar behavior to the {@link groupBy} operator with the
+ * difference that both groups are available immediately when using `splitBy`.
+ *
+ * ## Example
+ *
+ * ```ts
+ * import { interval } from 'rxjs';
+ * import { splitBy, mergeMap, mapTo, mergeAll, groupBy } from 'rxjs/operators';
+ *
+ * interval(1000).pipe(
+ *   splitBy((value) => value % 2 === 0),
+ *   mergeAll(),
+ *   mergeMap((bucket, index) => {
+ *     const kind = index === 0 ? 'even' : 'odd';
+ *     console.log(`${kind} bucket discovered`);
+ *
+ *     return bucket.pipe(mapTo(kind));
+ *   }),
+ * ).subscribe(console.log);
+ *
+ * // Result
+ * // "even bucket discovered"
+ * // "odd bucket discovered"
+ * // "even"
+ * // "odd"
+ * // "even"
+ * // "odd"
+ * // "even"
+ * // ...
+ *
+ * // groupBy is similar but different
+ *
+ * interval(1000).pipe(
+ *   groupBy((value) => value % 2 === 0),
+ *   mergeMap((group) => {
+ *     const kind = group.key ? 'even' : 'odd';
+ *     console.log(`${kind} group discovered`);
+ *
+ *     return group.pipe(mapTo(kind));
+ *   }),
+ * ).subscribe(console.log);
+ *
+ * // Result
+ * // "even group discovered"
+ * // "even"
+ * // "odd group discovered"
+ * // "odd"
+ * // "even"
+ * // "odd"
+ * // "even"
+ * // ...
+ *
+ * ```
+ *
+ * @param predicate The predicate to partition by
+ * @return A function that returns an Observable that does a single emit
+ *  containing the two observables for matching and non-matching values
+ *
+ * @see {@link buckets}
+ * @see {@link groupBy}
+ * @see {@link partition}
+ */
+export function splitBy<T>(
+  predicate: (value: T) => boolean,
+  options: SplitByOptions<T> = {}
+): OperatorFunction<T, [Observable<T>, Observable<T>]> {
```

Comment:
This function has overload signatures and they hide the implementation signature. So you'll need this to be an overload signature in its own right - even though the implementation is exactly the same. That's my understanding, at least.

Reply:
actually the options are just missing from the other signatures. typescript only shows the overloads to the callers not the impl signature. the only requirement for the impl signature is that is has to be assignable to each overload signature.

will fix the missing argument